metrics/cli/test/deploy/deployer_test.dart

```diff
@@ -0,0 +1,395 @@
+
+        await deployer.deploy();
+
+        verify(gitCommand.clone(
+          DeployConstants.repoURL,
+          DeployConstants.tempDir,
+        )).called(once);
+      },
+    );
+
+    test(
+      ".deploy() clones the git repository before building the flutter application",
+      () async {
+        whenGetDirectory().thenReturn(directory);
+        await deployer.deploy();
+
+        verifyInOrder([
+          gitCommand.clone(DeployConstants.repoURL, DeployConstants.tempDir),
+          flutterService.build(DeployConstants.webPath),
+        ]);
+      },
+    );
+
+    test(
+      ".deploy() builds the flutter application",
+      () async {
+        whenGetDirectory().thenReturn(directory);
+
+        await deployer.deploy();
+
+        verify(flutterService.build(DeployConstants.webPath)).called(once);
+      },
+    );
+
+    test(
+      ".deploy() sets the default Firebase project",
+      () async {
+        const projectId = 'testId';
+        const firebaseToken = 'testToken';
+        whenCreateGCloudProject().thenAnswer((_) => Future.value(projectId));
+        whenLoginToFirebase().thenAnswer((_) => Future.value(firebaseToken));
+        whenGetDirectory().thenReturn(directory);
+
+        await deployer.deploy();
+
+        verify(
+          firebaseCommand.setFirebaseProject(
+            projectId,
+            DeployConstants.webPath,
+            firebaseToken,
+          ),
+        ).called(once);
+      },
+    );
+
+    test(
+      ".deploy() sets the default Firebase project before applying the Firebase target",
+      () async {
+        whenGetDirectory().thenReturn(directory);
+        await deployer.deploy();
+
+        verifyInOrder([
+          firebaseCommand.setFirebaseProject(any, any, any),
+          firebaseCommand.applyTarget(any, any, any, any),
+        ]);
+      },
+    );
+
+    test(
+      ".deploy() applies the Firebase target",
+      () async {
+        const projectId = 'testId';
+        const firebaseToken = 'testToken';
+        whenCreateGCloudProject().thenAnswer((_) => Future.value(projectId));
+        whenLoginToFirebase().thenAnswer((_) => Future.value(firebaseToken));
+        whenGetDirectory().thenReturn(directory);
+
+        await deployer.deploy();
+
+        verify(
+          firebaseCommand.applyTarget(
+            projectId,
+            DeployConstants.firebaseTarget,
+            DeployConstants.webPath,
+            firebaseToken,
+          ),
+        ).called(once);
+      },
+    );
+
+    test(
+      ".deploy() applies the Firebase target before deploying to the hosting",
+      () async {
+        whenGetDirectory().thenReturn(directory);
+        await deployer.deploy();
+
+        verifyInOrder([
+          firebaseCommand.applyTarget(any, any, any, any),
+          firebaseCommand.deployHosting(any, any, any),
+        ]);
+      },
+    );
+
+    test(
+      ".deploy() deploys the target to the hosting",
```

Comment:
Would it be great to test that this method builds the Flutter application before deploying it to the hosting?